indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3244,6 +3275,40 @@ private void checkCurrentTaskState() throws ExecutionException, InterruptedExcep
     Futures.successfulAsList(futures).get(futureTimeoutInSeconds, TimeUnit.SECONDS);
   }
 
+  private void checkIfStreamInactiveAndTurnSupervisorIdle()
+  {
+    if (!spec.getSupervisorStateManagerConfig().isEnableIdleBehaviour()
+        || !spec.getIoConfig().isEnableIdleBehaviour()) {
+      return;
```

Comment:
Isn't this return premature? Shouldn't the following state assignment occur:
      stateManager.maybeSetState(SupervisorStateManager.BasicState.RUNNING);

Reply:
Since if this condition turns out to be the case, It's unlikely to change while Supervisor is running hence unlikely to be moved to Idle state. In case of spec update or cluster restart, it would restart the Supervisor as well then in such a scenario Idle state would be erased as well if I'm not missing something.

Reply:
Could you please add a check to see if it is suspended?
If not, a SUSPENDED supervisor could be shown as IDLE.
While this doesn't affect ingestion, it could be misleading to the user

Reply:
Addressed it in the latest commit, Thanks.